More SPEC CPU 2006 execution flexibility

cloudharmony opened the pull request:
I use Intel compiled binaries and custom config/macros for 32 and 64 bit execution, and different SSE flags. Currently the SPEC CPU benchmark is hardcoded for a GCC 4.7 config and does not allow setting of any macros or runtime settings. This pull request consists of the following changes:

1. Added --runspec_config argument to allow override of default
   configuration linux64-x64-gcc47.cfg

2. Added --runspec_iterations argument to allow test execution using
   custom iteration count (default is 3)

3. Added --runspec_define argument to enable proprocessor macros

4. Added --runspec_enable_32bit to enable 32-bit execution on 64-bit
   images - useful when CPU cores to GB memory ratio is less than 2GB
   Setting this flag will install multilib packages
   (packages/multilib.py)

5. Added numactl package to enable NUMA scheduling

voellm commented:
ok to test

voellm commented:
Jason - You ok fixing up the flake8 issues below?  If not I can also address them.

Here is how to get it setup:
-------------------------------------------------------------------------
Configuring a local development environment
This can optionally done within a virtualenv.

Clone the repository:
* git clone git@github.com:GoogleCloudPlatform/PerfKitBenchmarker.git -b dev && cd PerfKitBenchmarker
* Install Python 2.7.
Install pip:
* sudo apt-get install -y python-pip
* Install PerfKitBenchmarker python dependencies:
* sudo pip install -r requirements.txt
Install PerfKitBenchmarker test dependencies:
* sudo pip install -r test-requirements.txt
Install git pre-commit hooks, to enable linting and copyright checks:
* hooks/install.sh
----------------------------------------------------------------------

Run flake8 from the root directory (install with pip install -r test-requirements.txt).

----------------------------------------------------------------------

GitHub pull request #230 of commit aa8f5980fe76ed433e2f8343721c4d3dbf9639f1 automatically merged.
 > git checkout -f 67ced1d7187e30fdcee6a53fba819b23bd86f67e
First time build. Skipping changelog.
+ flake8
./perfkitbenchmarker/packages/multilib.py:17:1: E302 expected 2 blank lines, found 1
def YumInstall(vm):
^
./perfkitbenchmarker/packages/multilib.py:21:1: E302 expected 2 blank lines, found 1
def AptInstall(vm):
^
./perfkitbenchmarker/packages/numactl.py:17:1: E302 expected 2 blank lines, found 1
def _Install(vm):
^
./perfkitbenchmarker/packages/numactl.py:21:1: E302 expected 2 blank lines, found 1
def YumInstall(vm):
^
./perfkitbenchmarker/packages/numactl.py:25:1: E302 expected 2 blank lines, found 1
def AptInstall(vm):
^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:37:1: W293 blank line contains whitespace

^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:38:63: W291 trailing whitespace
flags.DEFINE_string('runspec_config', 'linux64-x64-gcc47.cfg',
                                                              ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:41:1: W293 blank line contains whitespace

^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:42:46: W291 trailing whitespace
flags.DEFINE_integer('runspec_iterations', 3,
                                             ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:43:21: E128 continuation line under-indented for visual indent
                    'number of benchmark iterations to execute - default 3 '
                    ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:44:21: E128 continuation line under-indented for visual indent
                    '(runspec --iterations argument)')
                    ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:45:1: W293 blank line contains whitespace

^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:46:42: W291 trailing whitespace
flags.DEFINE_string('runspec_define', '',
                                         ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:50:1: W293 blank line contains whitespace

^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:248:67: W291 trailing whitespace
                   % (vm.spec_dir, FLAGS.runspec_config, num_cpus,
                                                                  ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:249:20: E128 continuation line under-indented for visual indent
                   ' --iterations=' + `FLAGS.runspec_iterations` if FLAGS.runspec_iterations != 3 else '',
                   ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:249:39: W604 backticks are deprecated, use 'repr()'
                   ' --iterations=' + `FLAGS.runspec_iterations` if FLAGS.runspec_iterations != 3 else '',
                                      ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:249:81: E501 line too long (106 > 80 characters)
                   ' --iterations=' + `FLAGS.runspec_iterations` if FLAGS.runspec_iterations != 3 else '',
                                                                                ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:250:20: E128 continuation line under-indented for visual indent
                   ' --define ' + ' --define '.join(FLAGS.runspec_define.split(',')) if FLAGS.runspec_define != '' else '',
                   ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:250:81: E501 line too long (123 > 80 characters)
                   ' --define ' + ' --define '.join(FLAGS.runspec_define.split(',')) if FLAGS.runspec_define != '' else '',
                                                                                ^
./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:251:20: E128 continuation line under-indented for visual indent
                   FLAGS.benchmark_subset))
                   ^

cloudharmony commented:
I’ll fix this.

> On Apr 28, 2015, at 5:26 PM, Anthony Voellm aka Tony the P3rfguy <notifications@github.com> wrote:
> 
> Jason - You ok fixing up the flake8 issues below? If not I can also address them.
> 
> Here is how to get it setup:
> 
> Configuring a local development environment
> This can optionally done within a virtualenv.
> 
> Clone the repository:
> 
> git clone git@github.com <mailto:git@github.com>:GoogleCloudPlatform/PerfKitBenchmarker.git -b dev && cd PerfKitBenchmarker
> Install Python 2.7. Install pip:
> sudo apt-get install -y python-pip
> Install PerfKitBenchmarker python dependencies:
> sudo pip install -r requirements.txt Install PerfKitBenchmarker test dependencies:
> sudo pip install -r test-requirements.txt Install git pre-commit hooks, to enable linting and copyright checks:
> * hooks/install.sh
> 
> Run flake8 from the root directory (install with pip install -r test-requirements.txt).
> 
> GitHub pull request #230 <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/230> of commit aa8f598 <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/commit/aa8f5980fe76ed433e2f8343721c4d3dbf9639f1> automatically merged.
> 
> git checkout -f 67ced1d <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/commit/67ced1d7187e30fdcee6a53fba819b23bd86f67e>
> First time build. Skipping changelog.
> 
> flake8 ./perfkitbenchmarker/packages/multilib.py:17:1: E302 expected 2 blank lines, found 1 def YumInstall(vm): ^ ./perfkitbenchmarker/packages/multilib.py:21:1: E302 expected 2 blank lines, found 1 def AptInstall(vm): ^ ./perfkitbenchmarker/packages/numactl.py:17:1: E302 expected 2 blank lines, found 1 def _Install(vm): ^ ./perfkitbenchmarker/packages/numactl.py:21:1: E302 expected 2 blank lines, found 1 def YumInstall(vm): ^ ./perfkitbenchmarker/packages/numactl.py:25:1: E302 expected 2 blank lines, found 1 def AptInstall(vm): ^ ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:37:1: W293 blank line contains whitespace
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:38:63: W291 trailing whitespace
> flags.DEFINE_string('runspec_config', 'linux64-x64-gcc47.cfg',
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:41:1: W293 blank line contains whitespace
> 
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:42:46: W291 trailing whitespace
> flags.DEFINE_integer('runspec_iterations', 3,
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:43:21: E128 continuation line under-indented for visual indent
> 'number of benchmark iterations to execute - default 3 '
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:44:21: E128 continuation line under-indented for visual indent
> '(runspec --iterations argument)')
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:45:1: W293 blank line contains whitespace
> 
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:46:42: W291 trailing whitespace
> flags.DEFINE_string('runspec_define', '',
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:50:1: W293 blank line contains whitespace
> 
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:248:67: W291 trailing whitespace
> % (vm.spec_dir, FLAGS.runspec_config, num_cpus,
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:249:20: E128 continuation line under-indented for visual indent
> ' --iterations=' + FLAGS.runspec_iterations if FLAGS.runspec_iterations != 3 else '',
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:249:39: W604 backticks are deprecated, use 'repr()'
> ' --iterations=' + FLAGS.runspec_iterations if FLAGS.runspec_iterations != 3 else '',
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:249:81: E501 line too long (106 > 80 characters)
> ' --iterations=' + FLAGS.runspec_iterations if FLAGS.runspec_iterations != 3 else '',
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:250:20: E128 continuation line under-indented for visual indent
> ' --define ' + ' --define '.join(FLAGS.runspec_define.split(',')) if FLAGS.runspec_define != '' else '',
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:250:81: E501 line too long (123 > 80 characters)
> ' --define ' + ' --define '.join(FLAGS.runspec_define.split(',')) if FLAGS.runspec_define != '' else '',
> ^
> ./perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py:251:20: E128 continuation line under-indented for visual indent
> FLAGS.benchmark_subset))
> ^
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/230#issuecomment-97271371>.
>

cloudharmony commented:
Do you need anything else to merge?

voellm commented:
I must have missed the updates.  I'll review first thing in the morning.

On Mon, May 4, 2015 at 4:01 PM, Jason Read <notifications@github.com> wrote:

> Do you need anything else to merge?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/230#issuecomment-98880527>
> .
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/

voellm commented:
ok to test

voellm commented:
test this please

voellm commented:
FYI "ok to test" 'test this please" are messages to our test bot.

voellm commented:
I have verified the CLA is signed.  Thank You!
